src/utils.spec.tsx

```diff
@@ -188,4 +195,46 @@ describe(__filename, () => {
       expect(getPathFromQueryString(history)).toEqual(null);
     });
   });
+
+  describe('createCodeLineAnchorGetter', () => {
+    it('returns getCodeLineAnchor if compareInfo is null', () => {
+      expect(createCodeLineAnchorGetter(null)).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo is undefined', () => {
+      expect(createCodeLineAnchorGetter(undefined)).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo.diff is falsey', () => {
+      expect(
+        createCodeLineAnchorGetter({ diff: null, mimeType: 'mime/type' }),
+      ).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor from ForwardComparisonMap is compareInfo.diff exists', () => {
+      const compareInfo = createFakeCompareInfo();
+      if (!compareInfo.diff) {
+        throw new Error('compareInfo.diff was unexpectedly empty');
+      }
+      const map = new ForwardComparisonMap(compareInfo.diff);
+
+      const getterFromFactory = createCodeLineAnchorGetter(compareInfo);
+      const getterFromMap = map.createCodeLineAnchorGetter();
+
+      // Generate a getter without compareInfo to verify that it returns a
+      // different result.
+      const getterFromFactoryWithoutCompareInfo = createCodeLineAnchorGetter(
+        null,
+      );
+
+      // We cannot use `toEqual` for the two getters above because they are
+      // different instances, so we check that they return the same thing.
+      expect(getterFromFactory(1)).toEqual(getterFromMap(1));
+      // And verify that a getter from the factory without compareInfo does not
+      // return the same thing.
+      expect(getterFromFactoryWithoutCompareInfo(1)).not.toEqual(
```

Comment:
I think it would be fine to just check that the getter functions are different. It accomplishes the same thing and is slightly more readable to me.

```diff
diff --git a/src/utils.spec.tsx b/src/utils.spec.tsx
index 2ff167c..f56efa4 100644
--- a/src/utils.spec.tsx
+++ b/src/utils.spec.tsx
@@ -232,9 +232,7 @@ describe(__filename, () => {
       expect(getterFromFactory(1)).toEqual(getterFromMap(1));
       // And verify that a getter from the factory without compareInfo does not
       // return the same thing.
-      expect(getterFromFactoryWithoutCompareInfo(1)).not.toEqual(
-        getterFromMap(1),
-      );
+      expect(getterFromFactory).not.toEqual(getterFromFactoryWithoutCompareInfo);
     });
   });
 });
```

Reply:
I don't think that will work for the same reason I couldn't use an `isEqual`. Two objects from the factory are never equal to each other, so saying that `getterFromFactory` is not equal to `getterFromFactoryWithoutCompareInfo` does not prove anything. I tested this by creating a `getterFromFactory2` using the exact same call that generates `getterFromFactory`, and those two getters fail an `isEqual` assertion.

Reply:
The test I proposed is just to make sure that [this line](https://github.com/mozilla/addons-code-manager/blob/df3b815b5d07b16790696bdb31d4fda8ff96bc4c/src/utils.tsx#L76) never gets hit. I still think that would be a sufficient test here. Either way works, though.

Reply:
Ah, right, I was thinking about it backwards. That makes sense now. I'll change it.